src/parser/demonhunter/havoc/modules/talents/Demonic.tsx

```diff
@@ -42,12 +45,13 @@ class Demonic extends Analyzer {
     this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.DEATH_SWEEP), this.onDeathSweepCast);
   }
 
-  onEyeBeamCast(event) {
+  onEyeBeamCast(event: CastEvent) {
     const hasMetaBuff = this.selectedCombatant.hasBuff(SPELLS.METAMORPHOSIS_HAVOC_BUFF.id, event.timestamp - 1000);
 
-    if (hasMetaBuff) {
+    if (hasMetaBuff || !(this.selectedCombatant.hasTalent(SPELLS.TRAIL_OF_RUIN_TALENT) || this.selectedCombatant.hasTalent(SPELLS.FIRST_BLOOD_TALENT))) {
```

Comment:
since talents can't change throughout the fight i'd put these two calls in the constructor and make them global variables

Reply:
Moved it into `talentsCheck`, line 31. If this is what u meant by moving them into the constructor?